datafusion/common/src/utils/mod.rs

```diff
@@ -342,6 +342,20 @@ pub fn array_into_list_array(arr: ArrayRef, nullable: bool) -> ListArray {
     )
 }
 
+pub fn array_into_list_array_with_field_name(
```

Comment:
So is the idea that we will backport this to 43.0.0 and then consolidate the other functions on main? That makes sense to me if so